src/drivers/bitbucket_cloud.js

```diff
@@ -0,0 +1,79 @@
+const fetch = require('node-fetch');
+const FormData = require('form-data');
+const { URL, URLSearchParams } = require('url');
+const {Base64} = require('js-base64');
+
+const { fetch_upload_data } = require('../utils');
+
+class BitBucketCloud {
+  constructor(opts = {}) {
+    const { repo, token } = opts;
+
+    if (!token) throw new Error('token not found');
+    if (!repo) throw new Error('repo not found');
+
+    this.token = token;
+    this.repo = repo;
+
+    const { protocol, host, pathname } = new URL(this.repo);
+    this.repo_origin = `${protocol}//${host}`;
+    this.api = 'https://api.bitbucket.org/2.0';
+    this.project_path = encodeURIComponent(pathname.substring(1));
+  }
+
+  async comment_create(opts = {}) {
+    const { project_path } = this;
+    const { commit_sha, report} = opts;
+
+    // Print some variables
+    const endpoint = `/repositories/${project_path}/commit/${commit_sha}/comments/`;
+    const body= JSON.stringify({"content": {"raw": report}});
+
+    const output = await this.request({ endpoint, method: 'POST', body });
+
+    return output;
+  }
+
+  async check_create() {
+    throw new Error('BitBucket Cloud does not support check!');
+  }
+
+  async upload(opts = {}) {
+    throw new Error('BitBucket Cloud does not support upload!');
+  }
+
+  async runner_token() {
+    throw new Error('BitBucket Cloud does not support runner_token!');
+  }
+
+  async register_runner(opts = {}) {
+    throw new Error('BitBucket does not support register_runner!');
+  }
+
+  async unregister_runner(opts = {}) {
+    throw new Error('BitBucket does not support unregister_runner!');
+  }
+
+  async runner_by_name(opts = {}) {
+    throw new Error('BitBucket does not support runner_by_name!');
+  }
+
+  async request(opts = {}) {
+    const { token, api } = this;
+    const { endpoint, method = 'GET', body } = opts;
+
+    if (!endpoint) throw new Error('BitBucket API endpoint not found');
+    const {BITBUCKET_WORKSPACE}=process.env;
```

Comment:
This makes bitbucket dependant on Env again. If we are in gitlab CI pushing to Bitbucket would need to setup this ENV variable previously.

Reply:
Hm, if they're in GitLab CI trying to make a request to the BBCloud api, we need some way of getting their BB username. 

We could try using the repo URL, but if they're using an organizational workspace the URL won't necessarily contain their account name. What makes sense?

Assuming we let folks add their username as a secret (when not using BBCloud as driver), I imagine we can do:
```
if (type(repo) == BBCloud){
 const {BITBUCKET_WORKSPACE}=process.env;
 const username = BITBUCKET_WORKSPACE
} else{
 const {BITBUCKET_ACCOUNT_SECRET} = process.env;
 const username = BITBUCKET_ACCOUNT_SECRET
};

```

Reply:
> we need some way of getting their BB username.

As input or pre calculated token see below.
We can't depend on variables. The new architecture tries to infer as much as it can but now this driver wont operate in Gitlab CI without setting and ENV var